src/react/components/common/cloudFilePicker/cloudFilePicker.tsx

```diff
@@ -121,8 +121,18 @@ export class CloudFilePicker extends React.Component<ICloudFilePickerProps, IClo
         />;
     }
 
+    private isCloudConnection(connection: IConnection): boolean {
```

Comment:
Nice - this can likely get even easier after I get my PR in.  We can add additional properties at time of registration so we can lookup the providers registration options without having to instantiate it.